tests/hydrostatic_compression.cc

```diff
@@ -0,0 +1,297 @@
+                        - 0.1e1 + (0.5e0 - z) * (0.1e1 - alpha * sin(0.2e1 * pi * x) * pi * sin(pi * z));
+
+
+            for (unsigned int i=0, i_stokes=0; i_stokes<stokes_dofs_per_cell; /*increment at end of loop*/)
+              {
+                const unsigned int component_index_i = fe.system_to_component_index(i).first;
+
+                if (component_index_i == introspection.component_indices.pressure
+                    ||
+                    (
+                      component_index_i >= introspection.component_indices.velocities[0]
+                      &&
+                      component_index_i <= introspection.component_indices.velocities[dim-1]
+                    ))
+                  {
+                    scratch.phi_u[i_stokes]   = scratch.finite_element_values[introspection.extractors.velocities].value (i,q);
+                    scratch.phi_p[i_stokes]   = scratch.finite_element_values[introspection.extractors.pressure].value (i, q);
+
+                    ++i_stokes;
+                  }
+                ++i;
+              }
+
+            const double JxW = scratch.finite_element_values.JxW(q);
+
+            for (unsigned int i=0; i<stokes_dofs_per_cell; ++i)
+              {
+                data.local_rhs(i) += (
+                                       (force_u * scratch.phi_u[i]
+                                        + pressure_scaling * force_p * scratch.phi_p[i])
+                                     )
+                                     * JxW;
+              }
+          }
+      }
+
+  };
+
+
+  template <int dim>
+  void add_force_assembler(const SimulatorAccess<dim> &simulator_access,
+                           Assemblers::Manager<dim> &assemblers)
+  {
+
+    Assemblers::Interface<dim> *assembler = new ForceAssembler<dim>();
+    assemblers.stokes_system.push_back(std_cxx11::unique_ptr<Assemblers::Interface<dim> >(assembler));
+  }
+}
+
+
+
+
+template <int dim>
+void signal_connector (aspect::SimulatorSignals<dim> &signals)
+{
+  signals.set_assemblers.connect (&aspect::add_force_assembler<dim>);
+}
+
+ASPECT_REGISTER_SIGNALS_CONNECTOR(signal_connector<2>,
+                                  signal_connector<3>)
+
+
+
+
+
+// explicit instantiations
+namespace aspect
+{
+  ASPECT_REGISTER_GRAVITY_MODEL(Gravity,
+                                "MyGravity",
+                                "")
+
+  ASPECT_REGISTER_MATERIAL_MODEL(TestMaterial,
+                                 "test material",
+                                 "")
+
+  ASPECT_REGISTER_POSTPROCESSOR(ConvergencePostprocessor,
+                                "error calculation",
+                                "A postprocessor that compares the numerical solution to the analytical "
+                                "solution derived for compressible mantle convection in a 2D box as described "
+                                "in the manuscript and reports the error.")
```

Comment:
Which manuscript? This test seems to have some kind of analytical solution, so it would be nice to have a bit more documentation about that.